### PR TITLE
Added new restart recovery test.

### PR DESCRIPTION
**Pull request by HristoStaykov**
Added test to verify the correct recovery of the system with 1 non-primary
replica performing State Transfer and the others performing View Change.
Moved the functionality to wait for replicas in ST to complete in a static method.

* **Problem Overview**  
Test scenario with multiple view changes and 1 non primary replica in state transfer.
* **Testing Done**  
N/A

### CODE REVIEW

**Comment by titusen** [comment]
LGTM!

File: tests/apollo/test_skvbc_restart_recovery.py [hunk 1]
```diff
@@ -414,20 +386,106 @@ async def test_recovering_of_primary_with_initiated_view_change(self, bft_networ
...
-                        while primary_last_stable != fetching_last_stable:
-                            primary_last_stable = await bft_network.metrics.get(primary, *key)
-                            fetching_last_stable = await bft_network.metrics.get(r, *key)
-                            log.log_message(f"primary_last_stable={primary_last_stable} fetching_last_stable={fetching_last_stable}")
-                            await skvbc.run_concurrent_ops(50)
-                            await trio.sleep(seconds=5)
+            await self._await_replicas_in_state_transfer(log, bft_network, skvbc, primary)
 
             await bft_network.wait_for_replicas_to_reach_at_least_view(replicas_ids=bft_network.all_replicas(), expected_view=view, timeout=60)
+
```
Comment (titusen) [inline tests/apollo/test_skvbc_restart_recovery.py:392]: LGTM!

**Review by titusen — APPROVED** [review]
LGTM

**Review by teoparvanov — APPROVED** [review]
Looks good, thanks!

**Pull request merged**